src/realm/object-store/impl/apple/keychain_helper.hpp

```diff
@@ -23,23 +23,34 @@
 
 #if REALM_PLATFORM_APPLE
 
+#include <realm/util/encryption_key.hpp>
+
 #include <optional>
 #include <string_view>
-#include <vector>
 
 namespace realm::keychain {
 
 // Get the stored encryption key for the metadata realm if one exists.
-std::optional<std::vector<char>> get_existing_metadata_realm_key(std::string_view app_id,
-                                                                 std::string_view access_group);
+std::optional<util::EncryptionKey> get_existing_metadata_realm_key(std::string_view app_id,
+                                                                   std::string_view access_group);
 // Create a new encryption key and store it in the keychain. Returns none if
 // the key could not be stored.
-std::optional<std::vector<char>> create_new_metadata_realm_key(std::string_view app_id,
-                                                               std::string_view access_group);
+std::optional<util::EncryptionKey> create_new_metadata_realm_key(std::string_view app_id,
+                                                                 std::string_view access_group);
 
 // Delete the encryption key for the metadata realm from the keychain.
 void delete_metadata_realm_encryption_key(std::string_view app_id, std::string_view access_group);
 
+namespace impl {
```

Comment:
The tests intentionally do not reuse the implementation's helpers and instead read from the keychain directly to better test the functionality.